mlir/lib/Dialect/Tosa/IR/TosaOps.cpp

```diff
@@ -256,14 +282,53 @@ static LogicalResult verifyConvOp(T op) {
     return failure();
   }
 
-  // Quantized type must have constructed the quantizationattr, and unquantized
-  // types should not have a quantizationattr.
-  if ((inputIsQuant && !op.getQuantizationInfo()) ||
-      (!inputIsQuant && op.getQuantizationInfo())) {
-    op.emitOpError("quantizationattr is required for quantized type, and not "
-                   "allowed for float type");
-    return failure();
+  // We require an explicit input zero point and weight zero point for i8
+  // convolution. It isn't invalid to provide zero points for other types but
+  // they will be ignored.
```

Comment:
This last sentence reads as though a floating point operation with `zp=100` wouldn't result in an error, yet there's a check to make sure it's zero below. The behaviour below seems correct according to the spec, so maybe worth updating the sentence here to align?